src/util/movinginterquartilemean.cpp

```diff
@@ -32,7 +32,7 @@ double MovingInterquartileMean::insert(double value) {
     }
 
     // If list was already full, delete the oldest value:
-    if (m_list.size() == m_iListMaxSize + 1) {
+    if (m_list.size() == static_cast<unsigned int>(m_iListMaxSize + 1)) {
```

Comment:
unsigned int is a third type, no one has. 
cast either the right part to std::size_t or the left part to int.